components/coding/SkillRatingsComponent.tsx

```diff
@@ -42,11 +45,64 @@ export default function SkillRatingsComponent(props) {
     },
   });
 
+  useEffect(() => {
+    if (skillRatings.length > 0) {
+      const unitNames = skillRatings.map((skill) => skill.unitName);
+      const sections = unitNames.filter(
+        (unitName, index, array) => array.indexOf(unitName) === index
+      );
+      setSections(sections);
+      setActiveTab(sections[0]);
```

Comment:
if you switch to the second tab and start changing the sliders, it automatically jumps back to the first tab because of this line... skillRatings is in the dependency array of this useEffect, so the hook runs any time its value changes

i think you could actually move this logic into the onCompleted of your query that fetches the skill ratings

Reply:
That doesn't seem to work, because the useEffect is needed in order to make sure skillRatings is populated before the sections are set.  I was able to get it to work by creating a hasRun state hook.  This will be set to false initially, and then the useEffect will check !hasRun and if it's false will run the useEffect hook logic which at the end will set the hasRun to true -- which will cause the useEffect to stop running when skillRatings changes.  

The skillRatings dependency is needed to retrigger a run, because it needs to run multiple times at the beginning to keep checking for a value in order to initially set the skillRatings.  Otherwise, I'm stuck with a "loading..." screen.

The useEffect is needed in the first place because if the logic is put in the onComplete, it will try to set the sections always before a skillRatings value is set, and I will get nothing -- only a save button. 

There def could be a better solution over adding the hasRun -- so I'm open to suggestions!

Reply:
I think this solution is pretty good actually! Nice thought process through this - I think you're getting a better grasp on the order of operations. 

There might be a way to incorporate a Promise in the onCompleted to work around the issues you mentioned but this definitely works and is likely less complicated.

I'd maybe call `hasRun` something more descriptive like `haveTabsLoaded` so it's more clear what it does.

Reply:
perfect!